src/System.Private.CoreLib/src/Internal/Runtime/Augments/RuntimeThread.cs

```diff
@@ -46,54 +68,225 @@ internal SafeWaitHandle[] GetWaitedSafeWaitHandleArray(int requiredCapacity)
             return _waitedSafeWaitHandles.Items;
         }
 
-        public static RuntimeThread Create(ThreadStart start) { throw null; }
-        public static RuntimeThread Create(ThreadStart start, int maxStackSize) { throw null; }
-        public static RuntimeThread Create(ParameterizedThreadStart start) { throw null; }
-        public static RuntimeThread Create(ParameterizedThreadStart start, int maxStackSize) { throw null; }
+        public static RuntimeThread Create(ThreadStart start) => new RuntimeThread(start, 0);
+        public static RuntimeThread Create(ThreadStart start, int maxStackSize) => new RuntimeThread(start, maxStackSize);
+
+        public static RuntimeThread Create(ParameterizedThreadStart start) => new RuntimeThread(start, 0);
+        public static RuntimeThread Create(ParameterizedThreadStart start, int maxStackSize) => new RuntimeThread(start, maxStackSize);
 
         public static RuntimeThread CurrentThread
         {
             get
             {
                 RuntimeThread currentThread = t_currentThread;
-                if (currentThread != null)
+                return t_currentThread ?? InitializeExistingThread();
+            }
+        }
+
+        // Slow path executed once per thread
+        private static RuntimeThread InitializeExistingThread()
+        {
+            var currentThread = new RuntimeThread();
+            currentThread._managedThreadId = System.Threading.ManagedThreadId.GetCurrentThreadId();
+            Debug.Assert(currentThread._threadState == (int)ThreadState.Unstarted);
+            // The main thread is foreground, other ones are background
+            if (currentThread._managedThreadId.Id == System.Threading.ManagedThreadId.IdMainThread)
+            {
+                currentThread._threadState = (int)(ThreadState.Running);
```

Comment:
Environment.Exit terminates an application immediately, without waiting for foreground threads to finish their execution.